### DISCUSSION TITLE
Is it possible to see what NextJS pages exist inside of middleware?

### DISCUSSION THREAD

**Discussion by cmaycumber** [vercel/next.js#43642] [Help]
For some context, I’m currently trying to use Framer.com for my marketing site for a no-code building experience. I’d like to proxy all requests to Framer when a page doesn’t exist in NextJS using a rewrite. 

For this to work effeciently it would be helpful if I could detect whether or not the page was going to 404 (or that the page doesn’t exist) in NextJS then defer to the proxy to the framer site. 

Is this possible?

**Comment by icyJoseph** [comment]
Feels like the `next.config.js` rewrites should do the job here, https://nextjs.org/docs/api-reference/next.config.js/rewrites#incremental-adoption-of-nextjs:

> fallback rewrites are checked/applied, these are applied before rendering the 404 page and after dynamic routes/all static assets have been checked. If you use [fallback: true/'blocking'](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-true) in getStaticPaths, the fallback rewrites defined in your next.config.js will not be run.
